support/server_interceptor.h is not possible to include on its own

### What version of gRPC and what language are you using?
v1.28.x branch, but also true in master, C++.

### What operating system (Linux, Windows,...) and version?
Linux RHEL 7.5

### What runtime / compiler are you using (e.g. python version or version of gcc)
gcc 9.2

### What did you do?
Create a cpp file with contents:
```c++
#include <grpcpp/support/server_interceptor.h>
```

### What did you expect to see?
Successful compilation.

### What did you see instead?
```
In file included from /deps/include/grpcpp/impl/codegen/server_interceptor.h:25,
                 from /deps/include/grpcpp/support/server_interceptor.h:22,
                 from <path_to_file.cc>:
/deps/include/grpcpp/impl/codegen/interceptor.h:201:16: error: ‘unique_ptr’ in namespace ‘std’ does not name a template type
  201 |   virtual std::unique_ptr<ChannelInterface> GetInterceptedChannel() = 0;
      |                ^~~~~~~~~~
/deps/include/grpcpp/impl/codegen/interceptor.h:27:1: note: ‘std::unique_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
   26 | #include <grpcpp/impl/codegen/metadata_map.h>
  +++ |+#include <memory>
   27 |
```


### Anything else we should know about your project / environment?

--- a/include/grpcpp/impl/codegen/interceptor.h
+++ b/include/grpcpp/impl/codegen/interceptor.h
@@ -19,6 +19,8 @@
 #ifndef GRPCPP_IMPL_CODEGEN_INTERCEPTOR_H
 #define GRPCPP_IMPL_CODEGEN_INTERCEPTOR_H
 
+#include <memory>
+
 #include <grpc/impl/codegen/grpc_types.h>
 #include <grpcpp/impl/codegen/byte_buffer.h>
 #include <grpcpp/impl/codegen/config.h>

